manage4.js

```diff
@@ -286,14 +382,42 @@ export async function main(ns) {
                 }
             }
         } catch (e) {}
+
+        if (bool_manage_servers) {
+            manage_servers(ns, owned_servers)
+
+            for (var i of owned_servers) {
+                if (cfg.training == undefined) {
+                    throw "training not defined please tell script to use learning ";
+                }
+                if (i[1].waiting_to_sell)
+                    ns.killall(i[0])
+                else {
+
+                    if (!ns.fileExists("weaken_hard.script", i[0])) {

```

Comment:
This only needs to run inside the `if()` statement below -- we only need to copy the script if we will actually run it